Content/Items/Currency/CorruptShard.cs

```diff
@@ -0,0 +1,93 @@
+﻿using PathOfTerraria.Common.Enums;
+using PathOfTerraria.Common.Systems.Affixes;
+using PathOfTerraria.Common.Systems.Affixes.ItemTypes;
+using PathOfTerraria.Core.Items;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.Utilities;
+using GearItem = PathOfTerraria.Content.Items.Gear.Gear;
+
+namespace PathOfTerraria.Content.Items.Currency;
+
+internal class CorruptShard : CurrencyShard
+{
+	public override void SetStaticDefaults()
+	{
+		base.SetStaticDefaults();
+
+		ItemID.Sets.AnimatesAsSoul[Item.type] = true;
+		Main.RegisterItemAnimation(Item.type, new DrawAnimationVertical(5, 4));
+	}
+
+	public override void SetDefaults()
+	{
+		Item.CloneDefaults(ItemID.Silk);
+		Item.Size = new Vector2(30, 28);
+		Item.rare = ItemRarityID.Green;
+		Item.consumable = true; // Purely for the tooltip line
+	}
+
+	public override bool CanRightClick()
+	{
+		if (!Main.LocalPlayer.HeldItem.TryGetGlobalItem(out PoTGlobalItem _))
+		{
+			return false;
+		}
+
+		return !Main.LocalPlayer.HeldItem.GetInstanceData().Corrupted;
+	}
+
+	public override void RightClick(Player player)
+	{
+		if (Main.rand.NextBool(2)) // 50% chance to do nothing
+		{
+			return;
```

Comment:
We still need to set corrupted for the item no?
I only see it added in the AddAffix portion

Reply:
Shouldn't be able to attempt twice

Reply:
Didn't know that was the case. Will do.